### PR TITLE
[BUG] Despite different language setting: Ghostfolio only shows English after login #3802

### LINKED ISSUE #3802: [BUG] Despite different language setting: Ghostfolio only shows English after login
Hi everyone,

I have hosted Ghostfolio myself and have encountered a problem with the language setting. If you have selected a different language in the settings under "My Ghostfolio" (e.g. German), the English language is still displayed as the default after logging in.

**Steps to reproduce:**
1. host Ghostfolio yourself.
2. set the language in the settings to a language other than English (e.g. German).
3. open Ghostfolio in the browser and log in.

**Expected behavior:**
The selected language (e.g. German) should be used after login.

**Actual behavior:**
After logging in, English is displayed by default, even if another language has been selected in the settings. To use the correct language, you have to manually change the "/en/" in the URL in the browser address bar (e.g. to ".../de/home").

It would be great if this problem could be fixed so that the selected language is taken into account directly after login without having to manually adjust the URL.

Many thanks for the great work to Ghostfolio!

### PR DESCRIPTION
**Pull request by madhab-chandra-sahoo**
Fix #3802: Fetched logged-in user settings to display in the preferred language

[Language bug solved](https://www.awesomescreenshot.com/video/31967152?key=199fa44e710bf3de4f24803427ebf3ef)

### CODE REVIEW

**Comment by madhab-chandra-sahoo** [comment]
Hi @dtslvr, can you please review this PR?

File: apps/client/src/app/components/header/header.component.ts [hunk 1]
```diff
@@ -254,12 +256,23 @@ export class HeaderComponent implements OnChanges {
   }
 
   public setToken(aToken: string) {
+    console.log('Set Token called');
```
Comment (dtslvr) [inline apps/client/src/app/components/header/header.component.ts:259]: Please remove

File: apps/client/src/app/components/header/header.component.ts [hunk 2]
```diff
@@ -254,12 +256,23 @@ export class HeaderComponent implements OnChanges {
...
     this.tokenStorageService.saveToken(
       aToken,
       this.settingsStorageService.getSetting(KEY_STAY_SIGNED_IN) === 'true'
     );
-
-    this.router.navigate(['/']);
+    this.userService
+      .get(true)
```
Comment (dtslvr) [inline apps/client/src/app/components/header/header.component.ts:265]: `.get()`

File: apps/client/src/app/components/header/header.component.ts [hunk 3]
```diff
@@ -254,12 +256,23 @@ export class HeaderComponent implements OnChanges {
...
+      .pipe(takeUntil(this.unsubscribeSubject))
+      .subscribe((data) => {
```
Comment (dtslvr) [inline apps/client/src/app/components/header/header.component.ts:267]: `.subscribe((user) => {`

File: apps/client/src/app/components/header/header.component.ts [hunk 4]
```diff
@@ -254,12 +256,23 @@ export class HeaderComponent implements OnChanges {
...
+        const userLanguage = data.settings.language;
```
Comment (dtslvr) [inline apps/client/src/app/components/header/header.component.ts:268]: Replace the full block with

```
const userLanguage = user?.settings?.language;

if (userLanguage && document.documentElement.lang !== userLanguage) {
  window.location.href = `../${userLanguage}`;
} else {
  this.router.navigate(['/']);
}
```

File: apps/client/src/app/app.component.ts
Comment (dtslvr) [inline apps/client/src/app/app.component.ts:1]: Please revert all changes in this file (not needed)

File: apps/client/src/app/components/home-overview/home-overview.component.ts [hunk 5]
```diff
@@ -11,7 +11,6 @@ import {
   User
 } from '@ghostfolio/common/interfaces';
 import { hasPermission, permissions } from '@ghostfolio/common/permissions';
-import { DateRange } from '@ghostfolio/common/types';
```
Comment (dtslvr) [inline apps/client/src/app/components/home-overview/home-overview.component.ts:14]: 👀

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thanks for the contribution, @madhab-chandra-sahoo! 🙌🏻

I have added a few comments to align it with a similar implementation in the project (`UserAccountSettingsComponent`).

**Comment by dtslvr** [comment]
@madhab-chandra-sahoo Thanks for submitting the PR! I have reviewed it and left some comments with requested changes. Once those are addressed, feel free to ping me again, and I’ll be happy to take another look.

**Comment by madhab-chandra-sahoo** [comment]
@dtslvr will resolve the comments by tomorrow

File: apps/client/src/app/components/home-overview/home-overview.component.ts (hunk 5 above)
Comment (madhab-chandra-sahoo) [reply apps/client/src/app/components/home-overview/home-overview.component.ts:14]: It was unused, so I removed it.

**Comment by madhab-chandra-sahoo** [comment]
@dtslvr Please review the changes as suggested

**Pull request merged**